### DISCUSSION TITLE
[v2] How can I tell which window the event is triggered from in `on_menu_event`?

### DISCUSSION THREAD

**Discussion by percy507** [tauri-apps/tauri#7759] [Q&A] [answered]
How can I tell which window the event is triggered from in `on_menu_event`?

```rust
// On tauri v1, I can use event.window() to get the window.
.on_menu_event(move |event| {
  let win = event.window();
  win.emit(event.id(), "").ok();
})

// How can I do this in tauri v2?
.on_menu_event(move |app, event| {
  // let win = event.window(); // get error
  // win.emit(event.id(), "").ok();
})
```

**Comment by FabianLars** [comment, marked as answer]
Could you open a feature request (as a github issue) for this please? Not sure if this was an intentional change or just an oversight due to the huge refactor.

**Reply by percy507** [reply]
Ok. https://github.com/tauri-apps/tauri/issues/7768

**Reply by percy507** [reply]
In short.

In tauri v1, menu is bind with the whole app, so we need figure out the event is triggered from which window. While in tauri v2, menu is bind with the specific window.
